assertk-common/src/main/kotlin/assertk/assertions/comparable.kt

```diff
@@ -61,3 +61,14 @@ fun <A, B : Comparable<A>> Assert<B>.isStrictlyBetween(start: A, end: A) {
     if (actual > start && actual < end) return
     expected("to be strictly between:${show(start)} and ${show(end)} but was:${show(actual)}")
 }
+
+/**
+ * Asserts the value if it is close to the expected value with given delta.
+ */
+fun <T: Number> Assert<T>.isCloseTo(value: T, delta: T) {
```

Comment:
I agree that it very common for `float and doubles`. I thought as a `library` we can give for all `Number`(s), users can use it accordingly.

Thoughts? I am perfectly good to have this only for `float` and `double`

Reply:
So actually, don't think this will be correct for longs as converting them to a double is a narrowing conversion. https://stackoverflow.com/a/51915530

Reply:
Got it. Thanks. I will try to summerize
```kotlin
val longVariable = 4999999999999999999L
val doubleVariable = 5000000000000000000.0
assert(longVariable).isCloseTo(doubleVariable, delta = 0)
```
the above test should fail. But it is not because of `narrowing conversion`.

Hence we should not convert into `double`. In that case we have to write `isCloseTo` for all `primitive types`. May be we can stick to `Double` and `Float` now.

Is that okay?

Reply:
Added `isCloseTo` only for `Double` and `Float`.  And also change the error message like below

```kotlin
assertk.assert(10.1f).isCloseTo(15.0f, 3.5f)
// expected <10.1f> to be close to <15.0f> with delta of <3.5f>, but was not
```

Is that okay?

Reply:
Yeah, that looks good!